node_modules/col-rest/lib/rest/lti.js

```diff
@@ -103,7 +112,23 @@ module.exports = function(client) {
    * @param  {Response}       callback.response               The response object as returned by requestjs
    * @see col-lti/lib/rest.js for more information
    */
-  client.lti.whiteboardsLaunch = function(data, callback) {
-    client.request('/lti/whiteboards', 'POST', data, null, callback);
+  client.lti.whiteboardsLaunch = function(course, data, callback) {
+    var headers = {
+      'referer': getRefererHeader(course, 'whiteboards')
+    };
+    client.request('/lti/whiteboards', 'POST', data, headers, callback);
+  };
+
+  /**
+   * Get the referer header for a given course and tool
+   *
+   * @param  {Course}   course    The course in which the user is launching the LTI tool
+   * @param  {String}   tool      The tool that is being launched (One of `assetlibrary`, `whiteboards` or `engagementIndex`)
+   * @return {String}             A URL that can be used as the referer header for the launch of one of the LTI tools
+   * @api private
+   */
+  var getRefererHeader = function(course, tool) {
+    var protocol = (course.canvas.use_https) ? 'https' : 'http';
+    return protocol + '://' + course.canvas.canvas_api_domain + '/external_tools/' + tool;
```

Comment:
I'm still not entirely clear on how a `course.canvas.canvas_api_domain + '/external_tools/' + 'assetlibrary'` tool is being used as that results in an invalid Canvas URL